### PR TITLE
[CDAP-9092][CDAP-9093][CDAP-9273] New pipeline schedule modeless, with more basic scheduling options

### PR DESCRIPTION
**Pull request by tonybach**
JIRA: https://issues.cask.co/browse/CDAP-9273

This new schedule modeless comes with a new look, and allows for more basic options (e.g. every x hours instead of just every hour). All the basic options can also be converted to advanced options and vice versa, when the user saves the schedule, as long as the basic view can handle the user's cron expression.

Also removed the Validate button in Studio to make place for the Schedule button, as the former will be removed anyways as part of https://issues.cask.co/browse/CDAP-11550.

This PR will also resolve [CDAP-9092](https://issues.cask.co/browse/CDAP-9092) and [CDAP-9093](https://issues.cask.co/browse/CDAP-9093).

**TODO**:
- ~~Figure out how to set the "Max concurrent runs" input as part of the schedule config.~~ Will remove this for now, as this config is not implemented in the backend yet.
- ~~Figure out how to allow for editing the schedule after the pipeline has been deployed. Currently in detail view, editing is disabled, but the user can still start the schedule.~~ Talked with @bdmogal, we decided that we can keep this behavior for this PR, then will discuss later how/when we should implement the editing behavior.

### CODE REVIEW

File: cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+      this.advancedScheduleValues.min = cronValues[0];
+      this.advancedScheduleValues.hour = cronValues[1];
+      this.advancedScheduleValues.day = cronValues[2];
+      this.advancedScheduleValues.month = cronValues[3];
+      this.advancedScheduleValues.dayOfWeek = cronValues[4];
+
+      // Hourly
+      if (myHelpers.isNumeric(cronValues[0]) && cronValues[1].indexOf('/') !== -1 && cronValues[2] === '*' && cronValues[3] === '*' && cronValues[4] === '*') {
+        this.intervalOptionKey = 'Hourly';
+        this.hourlyOptions.startingAt = parseInt(cronValues[0]);
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js]: our convention is to use explicit radix for `parseInt`.

**Comment by tonybach** [comment]
Will also add commits for these newly added specs: https://issues.cask.co/browse/CDAP-9273?focusedCommentId=29755&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-29755

File: cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.html [hunk 2]
```diff
@@ -0,0 +1,418 @@
...
+    <div class="btn-group">
+      <a class="btn" ng-click="SchedulerCtrl.onClose()">
+        <span class="fa fa-remove"></span>
+      </a>
+    </div>
+  </div>
+  <div class="pipeline-scheduler-body">
+    <div class="schedule-content">
+      <fieldset ng-disabled="SchedulerCtrl._isDisabled">
+        <div class="schedule-types">
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.html]: should add a `row` class here. `col-*` classes are using `float: left`, so it needs to the clearfix, especially when the total number of column in the row is less than 12

File: cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+        this.yearlyOptions.startingAt.am_pm = convertedHour.format('A');
+        this.yearlyOptions.repeatEvery.day = parseInt(cronValues[2]);
+        this.yearlyOptions.repeatEvery.month = parseInt(cronValues[2]);
+      }
+
+      // Not handled in basic mode
+      else {
+        this.scheduleType = 'advanced';
+      }
+    }
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js]: Is it possible to extract this block to a service?

**Comment by elfenheart** [comment]
👍  Code LGTM

**Comment by bdmogal** [comment]
@tonybach text alignment is off:
image

**Comment by bdmogal** [comment]
@tonybach can we make this hour(s)?
image

**Comment by bdmogal** [comment]
Can we make this ``Configure Schedule for Pipeline "testschedule"``?
image

**Comment by bdmogal** [comment]
There is a 1px gap between the schedule button and the modeless that opens. I don't think that should exist. The same gap exists for the configure button in preview mode and pipeline detail view

image

**Comment by bdmogal** [comment]
@tonybach let's change this tooltip to:
``Please provide values for runtime arguments x, y, z, before running the pipeline``
image

**Comment by bdmogal** [comment]
@tonybach in preview mode, we're missing the Apply and Run method?
image

**Comment by bdmogal** [comment]
Also, I don't seem to be able to start preview after these changes.

**Comment by bdmogal** [comment]
@tonybach when you try to deploy a pipeline with an invalid schedule, you get an error saying "The cron expression <> is invalid". However, funnily, the pipeline still gets deployed (even though you stay in the studio). I found it because when you try to deploy again, it throws an error saying the pipeline exists:
image

In this pipeline, since the schedule is invalid, the schedule button is not shown at all:
image

If the UI gets an error that the schedule is invalid, the pipeline deployment should also fail.

**Comment by tonybach** [comment]
@bdmogal Regarding the configure modeless in preview, I thought we were following the same behavior in the pipeline detail view i.e. the config is for applying the runtime args only, and you have to click Run to actually run. I can change it to Apply & Run for preview, and keep Apply & Close for pipeline.

Also can you send me the pipeline that you were using to test preview?

**Comment by tonybach** [comment]
Regarding the cron expression error, have filed a backend JIRA: https://issues.cask.co/browse/CDAP-11578

**Review by bdmogal — APPROVED** [review]
LGTM

**Comment by tonybach** [comment]
Build: https://builds.cask.co/browse/CDAP-DRC6509-12

**Pull request merged**